policy-controller/src/admission.rs

```diff
@@ -170,6 +187,71 @@ fn parse_spec<T: DeserializeOwned>(req: AdmissionRequest) -> Result<(String, Str
     Ok((ns, name, spec))
 }
 
+#[async_trait::async_trait]
+impl Validate<AuthorizationPolicySpec> for Admission {
+    async fn validate(self, _ns: &str, _name: &str, spec: AuthorizationPolicySpec) -> Result<()> {
+        // TODO support namespace references?
+        if !spec.target_ref.targets_kind::<Server>() {
+            bail!(
+                "invalid targetRef kind: {}",
+                spec.target_ref.canonical_kind()
+            );
+        }
+
+        let mtls_authns_count = spec
+            .required_authentication_refs
+            .iter()
+            .filter(|authn| authn.targets_kind::<MeshTLSAuthentication>())
+            .count();
+        if mtls_authns_count > 1 {
+            bail!("only a single MeshTLSAuthentication may be set");
```

Comment:
This makes sense but was not immediately obvious. I don't think it's a lack of documentation, more just leaving this comment as an explanation to myself. In order for multiple identities to be authorized by an AuthorizationPolicy, we only can select at most one MeshTLSAuthentication. This MeshTLSAuthentication can select multiple identities though, so all those identities will be authorized clients to whatever the `targetRef` is.